src/main/java/edu/harvard/iq/dataverse/arquillianglassfishconfig/ArquillianGlassfishConfigurationParser.java

```diff
@@ -0,0 +1,95 @@
+package edu.harvard.iq.dataverse.arquillianglassfishconfig;
+
+import io.vavr.control.Try;
+import org.dom4j.Document;
+import org.dom4j.Element;
+import org.dom4j.Node;
+import org.dom4j.io.SAXReader;
+import org.dom4j.io.XMLWriter;
+
+import java.io.FileInputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * Class responsible for parsing glassfish configuration.
+ */
+public class ArquillianGlassfishConfigurationParser {
```

Comment:
This class do more than parsing.
I suggest renaming it to something like `ParametrizedGlassfishConfCreator`